node/index.js

```diff
@@ -445,10 +448,27 @@ function TChannelConnection(channel, socket, direction, remoteAddr) {
 
     socket.once('close', clearTimer);
 
-    self.socket
+    var stream = self.socket;
+
+    if (dumpEnabled) {
+        stream = stream.pipe(Spy(process.stdout, {
+            prefix: '>>> ' + self.remoteAddr + ' '
+        }));
+    }
+
+    stream = stream
         .pipe(self.reader)
         .pipe(self.handler)
         .pipe(self.writer)
+        ;
+
+    if (dumpEnabled) {
+        stream = stream.pipe(Spy(process.stdout, {
```

Comment:
The readable stream for `self.writer` does coalescing upto the high water mark so this might also not be whole frames.